### PR TITLE
Add browser docs

### PR DESCRIPTION
**Pull request by acolytec3**
This adds an initial readme for building for the browser.  

- Lists a few projects that use our code in browser
- Lists knoown dependency incompatibilities that require polyfills

### CODE REVIEW

File: config/BROWSER.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+- [Remix](https://remix.ethereum.org/) - a browser based Soldity compiler
+- [Blobs4Every1](https://github.com/acolytec3/blobs4every1) - a simple Vite + React web app that can generate Blob EIP 4844 Transactions and submit them to the Holesky chain
+
+## Potential Gotchas
+
+Some of our libraries use Javascript APIs where the NodeJS and browser versions are not 100% equivalent. In these cases, these APIs need to be polyfilled when building for the browser. Below are known potential browser incompatibilities and polyfills that are known to work in browser. Please refer to your particular bundler/framework on guidance for how to leverage these polyfills in your application.
+
+- `events` - [`eventemitter3`](https://www.npmjs.com/package/eventemitter3) - Our usage of the native NodeJS `eventemitter` class apparently onflicts with the typing of the browser equivalent `eventemitter` and polyfilling with `eventemitter3` is known to resolve this
+- `node:stream/web` - [`web-streams-polyfill`](https://www.npmjs.com/package/web-streams-polyfill) - The `trie` library uses a `ReadableStream` that requires this polyfill to work in browser
```
Comment (holgerd77) [inline config/BROWSER.md:18]: So this is now not needed any more after the releases.

File: config/BROWSER.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+- [TEVM](https://github.com/evmts/tevm-monorepo) - a JavaScript EVM client and a Solidity-to-TypeScript compiler
+- [Remix](https://remix.ethereum.org/) - a browser based Soldity compiler
+- [Blobs4Every1](https://github.com/acolytec3/blobs4every1) - a simple Vite + React web app that can generate Blob EIP 4844 Transactions and submit them to the Holesky chain
+
+## Potential Gotchas
+
+Some of our libraries use Javascript APIs where the NodeJS and browser versions are not 100% equivalent. In these cases, these APIs need to be polyfilled when building for the browser. Below are known potential browser incompatibilities and polyfills that are known to work in browser. Please refer to your particular bundler/framework on guidance for how to leverage these polyfills in your application.
+
+- `events` - [`eventemitter3`](https://www.npmjs.com/package/eventemitter3) - Our usage of the native NodeJS `eventemitter` class apparently onflicts with the typing of the browser equivalent `eventemitter` and polyfilling with `eventemitter3` is known to resolve this
```
Comment (holgerd77) [inline config/BROWSER.md:17]: Now that we are in discussion of also eliminate this major remaining gotcha, I wonder if there is enough substance left in the whole browser section to justify a separte document. 🤔

Basically only the examples left to stay for now.

Totally not against adding example projects, but maybe in this case we can move TEVM and Remix to the EVM, I would suggest a new section "External Projects" or "Using Projects" or similar in between here:

grafik

And Blobs4Every1 can go into the 4844 tx section of the tx library?

**Review by holgerd77 — COMMENTED** [review]
Some general comments with questions/suggestions on how to proceed here.

File: config/BROWSER.md (hunk 2 above)
Comment (acolytec3) [reply config/BROWSER.md:17]: I agree with this approach, provided we can make #3330 work.  Will leave this PR here for the moment until #3330 is closed or merged.

File: config/BROWSER.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Building for the browser
+
+One of the goals of our libraries is to be fully browser compatible. We have examples using the [`vite`](https://vitejs.dev/) framework in each of the libraries that can be referenced as a jumping off point.
+
+## Projects that use EthereumJS libraries in the browser
+
+Here are some projects that use our libraries in browser to give you some inspiration. These are not guaranteed to be 100% up to date but give you some examples of projects that use our code.
+
+- [TEVM](https://github.com/evmts/tevm-monorepo) - a JavaScript EVM client and a Solidity-to-TypeScript compiler
```
Comment (roninjin10) [inline config/BROWSER.md:9]: ❤️

**Comment by roninjin10** [comment]
an idea to add to this is to add[stack blitz](https://stackblitz.com/) example using ethereumjs in browser. Cool thing about that is you can get a forkable working example up and running in one click.

I did an example of what this looks like for OP here: https://github.com/ethereum-optimism/docs/pull/574

**Comment by acolytec3** [comment]
> an idea to add to this is to add[stack blitz](https://stackblitz.com/) example using ethereumjs in browser. Cool thing about that is you can get a forkable working example up and running in one click.
> 
> I did an example of what this looks like for OP here: https://github.com/ethereum-optimism/docs/pull/574

That's brilliant! I'm gonna have to try that out.

**Comment by holgerd77** [comment]
Will close this, so much changed along the way and not necessarily the best way to integrate docs any more.

**Pull request closed without merging**